src/compute/setFragmentAsInserted.ts

```diff
@@ -87,3 +90,56 @@ export function setFragmentAsInserted(inserted: Fragment, insertAttrs: NewInsert
   const updatedInserted = loopContentAndMergeText(inserted, insertAttrs, schema)
   return updatedInserted.length === 0 ? Fragment.empty : Fragment.fromArray(updatedInserted)
 }
+
+export function setFragmentAsWrapChange(inserted: Fragment, attrs: NewEmptyAttrs, schema: Schema) {
+  const content: PMNode[] = []
+
+  inserted.forEach((node) => {
+    content.push(
+      node.type.create(
+        {
+          ...node.attrs,
+          dataTracked: [addTrackIdIfDoesntExist(trackUtils.createNewWrapAttrs(attrs))],
+        },
+        setFragmentAsInserted(node.content, trackUtils.createNewInsertAttrs(attrs), schema),
+        node.marks
+      )
+    )
+  })
+
+  return Fragment.from(content)
+}
+
+/**
+ * Add split marker to first child and update last child to be as node split change
+ */
+export function setFragmentAsNodeSplit(inserted: Fragment, attrs: NewEmptyAttrs, schema: Schema) {
+  const firstChild = inserted.firstChild!,
+    lastChild = inserted.lastChild!
+  const splitMarkerId = uuidv4()
+  const splitMarker = schema.text('¶', [
```

Comment:
why do we need to add text? Can we just add that info to dataTracked as we normally do: to a mark for text nodes to attributes to blocknodes? If we split a node we modify 1 node and create a new one. It looks to me that should actually record a change not only on one of the nodes but on both of them. The donor node should have a record of part of its text being deleted. And the dataTracked id of this change can be used as a reference for markerID.
In that case we would consistently record changes for all nodes and we won't have to have special markers. It seems to me that it would be preferential not to have any special invisible characters in the content and especially those that are meaningful for the text flow, such as the pilcrow.

Reply:
> why do we need to add text? Can we just add that info to dataTracked as we normally do: to a mark for text nodes to attributes to blocknodes?

we need that text to know the exact location of the split point to return removed content back. what do you think about adding a leaf node just as a marker?

Reply:
if we keep text in the donor node will end up with duplicate content and it will be more complex to manage. 

**I returned back to your first suggestion** of adding just dataTracked attr to both nodes and on multiple splits for splitted or donor node will update dataTracked to reference new node